### ISSUE TITLE
Invalid synchronization: . Valid types are 'delta' and 'full'.

### ISSUE THREAD

**Issue by khertg** [milanmk/actions-file-deployer#12]
![image](https://user-images.githubusercontent.com/44987706/176870331-108f829a-a608-4440-bb97-6e0909433bb4.png)

**Comment by milanmk** [comment]
Please check if `sync` input value does not have leading/trailing space.

**Comment by milanmk** [comment]
Could you please share action log after enabling `debug` input parameter.

**Comment by milanmk** [comment]
@mattlag 
Thank you for providing action logs. It the same workflow did ran properly then I am guessing it was a temporary glitch with action runner.

**Comment by thomasmillercf** [comment]
@milanmk 

I have encountered this issue and found the cause ,well for my problem anyway. 
When trigging the deployment via a repo dispatch the following worked, however when trying to run the workflow manually, it would have the same issue as above.

The following is a cutdown example.

```
on:
  repository_dispatch:
    types: [production-release]
  workflow_dispatch:

.....

    - name: Deploy assets library to netstorage (CDN)
      uses: milanmk/actions-file-deployer@1.11
      with:
        remote-protocol: 'sftp'
        remote-host: ${{ inputs.cdnRemoteHost }}
        remote-user: 'sshacs'
        ssh-private-key: ${{ inputs.cdnPrivateKey }}
        remote-path: ${{ inputs.cdnRemotePath }}
        local-path: asset
        sync: full
        ssh-options: '-oHostKeyAlgorithms=+ssh-dss'
        ftp-options: 'set cmd:fail-exit true'
```


The root cause of my issue was that the input_sync expects the sync input to come from the workflow inputs when I manually trigger the workflow.


Basically my input was being overridden by the default input.sync
```
 input_sync=${{inputs.sync}}
        if [ "${{github.event_name}}" == "workflow_dispatch" ]; then
          input_sync=${{github.event.inputs.sync}}
fi
```

**Comment by milanmk** [comment]
You should have an input for workflow_dispatch like shown in the readme workflow example.

```
workflow_dispatch:
  inputs:
    sync:
      description: "File synchronization"
      required: true
      default: "delta"
```

**Comment by tswaehn** [comment]
I am trying this with manual trigger of the github action.

I do set sync in the settings:

```
      - name: sftp deploy
        uses: milanmk/actions-file-deployer@master
        with:
          remote-protocol: "sftp"
          remote-port: 22
          remote-host: ${{ secrets.DEPLOY_HOST }}
          remote-user: ${{ secrets.DEPLOY_USER }}
          remote-password: ${{ secrets.DEPLOY_PASS }}
          proxy: false
          sync: "full"
          local-path: "./public/*"
          remote-path: "test""
```

which leads to `Invalid synchronization: . Valid types are 'delta' and 'full'`

however when I additionally add sync to the input settings:
```
  workflow_dispatch:
    inputs:
      sync:
        description: "File synchronization"
        required: true
        default: "full"

```

it seems to work. 

**Why do I have to set sync twice?**

**Comment by milanmk** [comment]
@thomasmillercf 
I agree with your reasoning with having one way of defining the default. Have merged the changes in #44  and sorry for not approving your PR on time.
